pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/store/SegmentLocalFSDirectory.java

```diff
@@ -268,7 +267,7 @@ private synchronized void loadData()
       default:
         break;
     }
-    if (CollectionUtils.isNotEmpty(_segmentMetadata.getStarTreeV2MetadataList())) {
+    if (_segmentMetadata.getStarTreeV2MetadataList() != null) {
```

Comment:
looks like you've updated it to just handle an empty list instead?

Reply:
It cannot be empty, only null or non-empty. I modified this place to make it consistent with other callers